lib/widgets/message_list.dart

```diff
@@ -849,61 +852,75 @@ class MessageWithPossibleSender extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final message = item.message;
-    final time = _kMessageTimestampFormat
-      .format(DateTime.fromMillisecondsSinceEpoch(1000 * message.timestamp));
 
-    return GestureDetector(
-      behavior: HitTestBehavior.translucent,
-      onLongPress: () => showMessageActionSheet(context: context, message: message),
-      // TODO clean up this layout, by less precisely imitating web
-      child: Padding(
-        padding: const EdgeInsets.only(top: 2, bottom: 3, left: 8, right: 8),
-        child: Row(crossAxisAlignment: CrossAxisAlignment.start, children: [
-          item.showSender
-            ? Padding(
-                padding: const EdgeInsets.fromLTRB(3, 6, 11, 0),
+    final Widget? senderRow = item.showSender
+      ? Padding(
+          padding: const EdgeInsets.only(top: 2, bottom: 4, left: 16, right: 16),
+          child: Row(
```

Comment:
nit: This is a real long conditional expression. I think things get a bit cleaner if we instead use a normal `if` conditional:
```suggestion
    Widget? senderRow;
    if (item.showSender) {
      senderRow = Row(
```

(That version is combined with my other comment about keeping the `Padding` below.)

Then we can also leave out the `null` case; it's covered by the implicit initialization in `Widget? senderRow;`.

A follow-up bonus to using `if` rather than a conditional expression is that we can then have local variables within the `if` block to organize things. In particular, it's then possible to keep a local variable `time`, in the way I suggested in my original review:
https://github.com/zulip/zulip-flutter/pull/446#discussion_r1433448956
rather than inlining it and making the expression lengthier.